providers/implementations/keymgmt/mac_legacy_kmgmt.c

```diff
@@ -190,7 +190,7 @@ static int mac_key_fromdata(MAC_KEY *key, const OSSL_PARAM params[])
             return 0;
         }
         OPENSSL_secure_clear_free(key->priv_key, key->priv_key_len);
-        key->priv_key = OPENSSL_secure_malloc(p->data_size);
+        key->priv_key = OPENSSL_secure_malloc(p->data_size > 0 ? p->data_size : 1);
```

Comment:
A comment explaining this might be a good idea.

Reply:
Comment added